### DISCUSSION TITLE
How to help?

### DISCUSSION THREAD

**Discussion by Dumdidldum** [fnuecke/oc2#19] [General]
Hello

I would love to see this mod come to be!

Is there a way i could help?
I have programming experience, mostly with c# but also Java and done a few things with minecraft modding.

Do you have a discord where you talk about this?
Do you have a roadmap/design doc?

For example Crafting Recipes could be done by me, 
but then i would have to know what you have in mind...

**Comment by fnuecke** [comment]
Hi!

Thanks for your interest 👍🏻 

There's no official discord channel for this.
There's no road map.

But here's roughly where I'm planning on going with this:
- Work through the remaining open issues.
- While doing so, force myself to not add too many new ones.
- Figure out what to call it.
- Release it.
- Call it a day.

In particular I'd like to keep the "core" mod small. I'm open to the idea of an official companion mod that'll be home to extra stuff. But this I'd like to keep *relatively* compact.

@lucsoft is currently working on a first round for recipes (#14 ). Some extra feedback on recipes probably won't hurt. I'll probably have opinions, too, but I'll also note that I have not played MC in a long time, so I don't know how much those opinions will be worth.

After that there's the Fabric version. If you're up to helping out with that, that would be cool, too.

Another topic would be integration. If you have a good grasp on the modding landscape in 1.16.x and some ideas for which mods it would be nice to have some built-in support via specialized device types (e.g. something along the lines of the built-in providers for [capabilities defined by Forge](https://github.com/fnuecke/oc2/blob/1.16.4/src/main/java/li/cil/oc2/common/bus/device/provider/block/EnergyStorageBlockDeviceProvider.java)). I'm leaning towards putting that into its own mod, though. To make porting to newer versions easier by not having to wait for other mods to update (without having to comment out 90% of the integrations package).

Someone trying to do a small mod that uses the API would also be very enlightening. Some feedback on what's unclear, problematic, etc. would be great!

**Reply by atirutw** [reply]
> Call it a day.

That sounds like there will only be one release. Hopefully that's not the case.

**Comment by Emmixis** [comment]
Will there be a wiki/documentation that i can contribute to?

**Comment by fnuecke** [comment]
I'm not sure what the status on the Github wiki's permission system is at this point, do people still need to be contributors to edit it? If so, might set up a Github pages thing in the repo or so.

Either way, I want to wait a bit longer before setting this up. I'm not expecting major changes in what's there now, but I can't be 100% sure yet, and I'd hate to have to invalidate/throw away people's documentation work.
